packages/slider/src/HandleController.ts

```diff
@@ -0,0 +1,540 @@
+                    value=${model.value}
+                    aria-disabled=${ifDefined(
+                        this.host.disabled ? 'true' : undefined
+                    )}
+                    aria-valuetext=${this.formattedValueForHandle(model)}
+                    @change=${(event: Event) =>
+                        this.onInputChange(event, model)}
+                    @focus=${(event: Event) => this.onInputFocus(event, model)}
+                    @blur=${() => this.onInputBlur(model)}
+                />
+            </div>
+        `;
+    }
+
+    public render(): TemplateResult[] {
+        this.clearHandleComponentCache();
+        return this.model.map((model, index) => {
+            const zIndex = this.handleOrder.indexOf(model.name) + 1;
+            return this.renderHandle(model, index, zIndex);
+        });
+    }
+
+    /**
+     * Returns a list of track segment [start, end] tuples where the values are
+     * normalized to be between 0 and 1.
+     * @returns A list of track segment tuples [start, end]
+     */
+    public trackSegments(): [number, number][] {
+        const values = this.model.map((model) => model.normalizedValue);
+        values.sort((a, b) => a - b);
+
+        // The first segment always starts at 0
+        values.unshift(0);
+        return values.map((value, index, array) => [
+            value,
+            array[index + 1] ?? 1,
+        ]);
+    }
+
+    private updateModel(): void {
+        const handles = [...this.handles.values()];
+
+        const getRangeAndClamp = (index: number): RangeAndClamp => {
+            const handle = handles[index];
+            const previous = handles[index - 1];
+            const next = handles[index + 1];
+
+            const min = this.host.min as number;
+            const max = this.host.max as number;
+
+            const result: RangeAndClamp = {
+                range: { min: min, max: max },
+                clamp: { min: min, max: max },
+            };
+
+            if (handle.min === 'previous') {
+                if (previous) {
+                    for (let j = index - 1; j >= 0; j--) {
+                        const next = handles[j];
+                        if (typeof next.min === 'number') {
+                            result.range.min = next.min;
+                            break;
+                        }
+                    }
+                    result.clamp.min = Math.max(
+                        previous.value,
+                        result.range.min
+                    );
+                } else {
+                    /* c8 ignore next 3 */
+                    console.warn(
+                        'First slider handle cannot have attribute min="previous"'
+                    );
+                }
+            }
+            if (handle.max === 'next') {
+                if (next) {
+                    for (let j = index + 1; j < handles.length; j++) {
+                        const next = handles[j];
```

Comment:
JS doesn't mind using `next` here in different scopes, but it's a bit hard to track why we have next but then get a new next 🤔

Reply:
Agreed. I don't like variable shadowing either. Good catch...fixed.